specification/communication/data-plane/Microsoft.CommunicationServicesChat/preview/2020-11-01-preview3/communicationserviceschat.json

```diff
@@ -0,0 +1,1589 @@
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Error": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Error code",
+          "type": "string",
+          "readOnly": true
+        },
+        "message": {
+          "description": "Description of the error",
+          "type": "string",
+          "readOnly": true
+        },
+        "target": {
+          "description": "If applicable, would be used to indicate the property causing the error",
+          "type": "string",
+          "readOnly": true
+        },
+        "innerErrors": {
+          "description": "If applicable, inner errors would be returned for more details on the error",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Error"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "SendReadReceiptRequest": {
+      "description": "Request payload for sending a read receipt.",
+      "required": [
+        "chatMessageId"
+      ],
+      "type": "object",
+      "properties": {
+        "chatMessageId": {
+          "description": "Id of the latest chat message read by the user.",
+          "type": "string",
+          "example": "1592435762364"
+        }
+      }
+    },
+    "ChatMessagePriority": {
+      "description": "The chat message priority.",
+      "enum": [
+        "Normal",
+        "High"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ChatMessagePriority",
+        "modelAsString": true
+      }
+    },
+    "SendChatMessageRequest": {
+      "description": "Details of the message to send.",
+      "required": [
+        "content"
+      ],
+      "type": "object",
+      "properties": {
+        "priority": {
+          "$ref": "#/definitions/ChatMessagePriority"
+        },
+        "content": {
+          "description": "Chat message content.",
+          "type": "string",
+          "example": "Come one guys, lets go for lunch together."
+        },
+        "senderDisplayName": {
+          "description": "The display name of the chat message sender. This property is used to populate sender name for push notifications.",
+          "type": "string",
+          "example": "Bob Admin"
+        }
+      }
+    },
+    "SendChatMessageResult": {
+      "description": "Result of the send message operation.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "A server-generated message id.",
+          "type": "string",
+          "readOnly": true,
+          "example": "123456789"
+        }
+      }
+    },
+    "ChatMessage": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The id of the chat message. This id is server generated.",
+          "type": "string",
+          "readOnly": true,
+          "example": "123456789"
+        },
+        "type": {
```

Comment:
Let's also talk about potentially removing the thread activities from here. We should not really return the xml responses. Specially if they are being delivered through eventgrid and real time channel

Reply:
This is also a feature on its own. I initially thought also a message should be just a message (always), but after hearing some of the requirements from other P1 clients (dynamics / rave), for them a message includes all types (system level messages, text user messages, etc..), so we probably want to account for those before going ahead. 
There is also this underlying issue with pagination, so if we go with a separate end point, I think the right way of doing is is asking MFE to implement it, so we just expose it.